pkg/apply/applier.go

```diff
@@ -142,18 +154,130 @@ func (a *Applier) newStatusPoller() (poller.Poller, error) {
 	return polling.NewStatusPoller(c, mapper), nil
 }
 
+// previousInventories returns the infos for all the inventory objects
+// in the cluster.
 func (a *Applier) previousInventories(label, namespace string) ([]*resource.Info, error) {
 	return a.PruneOptions.RetrievePreviousInventoryObjects(label, namespace)
 }
 
+// infoHelperFactory returns a new instance of the InfoHelper.
+func (a *Applier) infoHelperFactory() info.InfoHelper {
+	return info.NewInfoHelper(a.factory, a.ApplyOptions.Namespace)
+}
+
+// readObjects reads the manifests from disk with the local flag set on
+// the Builder. This means that the resulting Info resources does not have
+// the mapping and client properties set. We are doing it this way because
+// it allows us to read in both CRDs and CRs in the same set without
+// errors.
+func (a *Applier) readObjects() ([]*resource.Info, error) {
+	r := a.factory.NewBuilder().
+		Local().
+		Unstructured().
+		Schema(a.ApplyOptions.Validator).
+		ContinueOnError().
+		NamespaceParam(a.ApplyOptions.Namespace).DefaultNamespace().
+		FilenameParam(a.ApplyOptions.EnforceNamespace, &a.ApplyOptions.DeleteOptions.FilenameOptions).
+		LabelSelectorParam(a.ApplyOptions.Selector).
+		Flatten().
+		Do()
+	if err := r.Err(); err != nil {
+		return nil, err
+	}
+
+	var infos []*resource.Info
+	err := r.Visit(func(info *resource.Info, err error) error {
```

Comment:
I don't think all of this using "Visit" is necessary. I think it's possible to just call "r.Infos()" to return the infos.

Reply:
Yeah, you are right. This was something that was left over from a previous iteration on this. Fixed now.